auth.php

```diff
@@ -243,19 +243,16 @@ public function canDo($cap) {
   }
 
   public function logIn() {
-    global $QUERY;
-    $login_url = DOKU_URL . 'doku.php?id=' . $QUERY;
+    $login_url = DOKU_URL . 'doku.php?id=' . $_GET["id"];
     phpCAS::setFixedServiceURL($login_url);
     phpCAS::forceAuthentication();
   }
 
   public function logOff() {
-    global $QUERY;
-
     if($this->_getOption('handlelogoutrequest')) { // dokuwiki + cas logout
       @session_start();
       session_destroy();
-      $logout_url = DOKU_URL . 'doku.php?id=' . $QUERY;
+      $logout_url = DOKU_URL . 'doku.php';
```

Comment:
```suggestion
      global $ID;
      $logout_url = wl($ID,'',true);
```

Why are you removing id parameter here?

Reply:
I thought it isn't needed and I managed to create a redirect loop yesterday, but that was due to something different.
So your solution is, once again, better.